Framework/DataHandling/src/LoadEmptyInstrument.cpp

```diff
@@ -53,16 +75,18 @@ int LoadEmptyInstrument::confidence(Kernel::FileDescriptor &descriptor) const {
 
 /// Initialisation method.
 void LoadEmptyInstrument::init() {
+  const std::vector<std::string> extensions{".xml", ".nxs", ".hdf5"};
   declareProperty(
       make_unique<FileProperty>("Filename", "", FileProperty::OptionalLoad,
-                                ".xml"),
+                                extensions),
       "The filename (including its full or relative path) of an instrument "
       "definition file. The file extension must either be .xml or .XML when "
-      "specifying an instrument definition file. Note Filename or "
+      "specifying an instrument definition file. Files can also be .hdf5 or "
+      ".nxs for usage with NeXus Geometry files. Note Filename or "
       "InstrumentName must be specified but not both.");
-  declareProperty(
-      "InstrumentName", "",
-      "Name of instrument. Can be used instead of Filename to specify an IDF");
+  declareProperty("InstrumentName", "",
+                  "Name of instrument. Can be used instead of Filename to "
+                  "specify an IDF");
```

Comment:
No have not thought that far ahead. For the time being, specifying an `InstrumentName` is tied to loading via the IDF route. In principle, this could be made to work in an identical way to the current route, but with a NeXus file with only the geometry aspects supplying the geometry. However, I believe for the ESS at least, there is a strong desire to always provide the consistent geometry alongside the data itself. i.e. move away from separate or inconsistent (parameter maps make IDF strings inconsistent) definitions.

Reply:
Neverhteless we need to support viewing an instrument without having data?

Reply:
Yes, we do. I would prefer that this was done via the aforementioned converter, with users still writing IDFs as the entry point for the time being. I don't like the `InstrumentName` parameter as which instrument definition is really being loaded is opaque and this has caused real problems in the past for users so would rather not apply that to the NeXus files if possible.